tasks/check_html_fixtures.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: false
+require 'diff/lcs'
+require 'diff/lcs/hunk'
+require 'fileutils'
+require "htmlbeautifier"
+
+
+# Get top-level Ruby examples
+file_paths = Dir["examples/*.rb"]
+
+# Get each filename
+file_names = file_paths.map do |file|
+  File.basename(file)
+end
+
+collection_of_html = []
+any_failures = false
+failures = []
+
+# ASCII color and format codes
+red = "\e[31m"
+green = "\e[32m"
+bold = "\e[1m"
+reset = "\e[0m"
+
+def diff_as_string(data_new, data_old, file_name)
+  data_old = data_old.split(/\n/).map! { |e| e.chomp }
+  data_new = data_new.split(/\n/).map! { |e| e.chomp }
+
+  diffs = Diff::LCS.diff(data_old, data_new)
+  return if diffs.empty?
+
+  context_lines = 1
+
+  oldhunk = hunk = nil
+  file_length_difference = 0
+
+  output = ""
+  diffs.each do |piece|
+    begin
+      hunk = Diff::LCS::Hunk.new(data_old, data_new, piece,
+                context_lines,
+                file_length_difference
+                )
+      file_length_difference = hunk.file_length_difference
+
+      next unless oldhunk
+
+      # Hunks may overlap, which is why we need to be careful when our
+      # diff includes lines of context. Otherwise, we might print
+      # redundant lines.
+      if (context_lines > 0) and hunk.overlaps?(oldhunk)
+        hunk.unshift(oldhunk)
+      else
+        output << oldhunk.diff(:unified, "#{file_name}:") << "\n"
+      end
+    ensure
+      oldhunk = hunk
+    end
+  end
+
+  #Handle the last remaining hunk
+  output << oldhunk.diff(:unified, "#{file_name}:") << "\n"
+end
+
+file_names.each do |file_name|
+  output = ""
+  pid = nil
+  command = "bundle exec ./exe/scarpe examples/#{file_name} --dev --debug"
```

Comment:
Ah, you're just running with --debug and reading the HTML off the console. Devious!

Reply:
I built this PR one step at a time starting with that. debug one example. multiple examples. save output. make it a task. compare. make output pretty. put in ci. , iterate, iterate, iterate, to what we have.